pulsar-client/src/main/java/org/apache/pulsar/client/impl/schema/AutoConsumeSchema.java

```diff
@@ -43,73 +44,82 @@
 @Slf4j
 public class AutoConsumeSchema implements Schema<GenericRecord> {
 
-    private Schema<?> schema;
+    private final Map<SchemaVersion, Schema<?>> schemaMap = new HashMap<>();
 
     private String topicName;
 
     private String componentName;
 
     private SchemaInfoProvider schemaInfoProvider;
 
+    public AutoConsumeSchema() {
+        schemaMap.put(BytesSchemaVersion.of(new byte[0]), Schema.BYTES);
```

Comment:
why do we need to assume that an empty schema version is "BYTES"  ? is this some kind of placeholder ? in this case we should use a constant and also explain in a comment

Reply:
The `Schema.BYTES` will not be uploaded to the broker and store in the schema storage, if the schema version in the message metadata is empty byte[], it means its schema is `Schema.BYTES`. I'll add this in the `schemaMap` init method and add a comment.